python/TestHarness/testers/ImageDiff.py

```diff
@@ -0,0 +1,78 @@
+import os
+import sys
+
+from RunPythonApp import RunPythonApp
+from mooseutils.ImageDiffer import ImageDiffer
+
+class ImageDiff(RunPythonApp):
+
+
+    @staticmethod
+    def validParams():
+        params = RunPythonApp.validParams()
+        params.addRequiredParam('imagediff', [], 'A list of files to compare against the gold.')
+        params.addParam('gold_dir', 'gold', "The directory where the \"golden standard\" files reside relative to the TEST_DIR: (default: ./gold/)")
+        params.addParam('allowed', 0.98, "Absolute zero cutoff used in exodiff comparisons.")
+        params.addParam('delete_output_before_running', True, "Delete pre-existing output files before running test. Only set to False if you know what you're doing!")
+        params.addParam('allowed_linux', "Absolute zero cuttoff used for linux machines, if not provided 'allowed' is used.")
+        params.addParam('allowed_darwin', "Absolute zero cuttoff used for Mac OS (Darwin) machines, if not provided 'allowed' is used.")
+        return params
+
+    def __init__(self, name, params):
+        RunPythonApp.__init__(self, name, params)
+
+    def prepare(self, options):
+        """
+        Cleans up image files from previous execution
+        """
+        if self.specs['delete_output_before_running'] == True:
```

Comment:
We've abstracted this method into the base class. See the implementations in ExoDiff and CSVDiff classes.